File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.cpp
#include <string>
#include <vector>
#include<set>
#include<map>
using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;
        set<int> sArr;
    map<int, int> arr;

    for (auto n : numbers)
    {
        arr[n]++;
    }

    for (auto it = arr.begin(); it != arr.end(); it++)
    {
        if (it->second != 1) sArr.insert(it->first*2);
        auto itt = it;
        itt++;
        for (; itt != arr.end(); itt++)
        {
            sArr.insert(it->first + itt->first);
        }
    }
    for (auto s : sArr)
    {
        answer.push_back(s);
    }
    return answer;
}

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.cs
using System;
using System.Linq;
using System.Collections.Generic;
public class Solution {
    public int[] solution(int[] numbers) {
        List<int> answer = new List<int>();

        for(int i = 0; i<numbers.Length;i++)
        {
            for(int j = i+1;j<numbers.Length;j++)
            {
                int res = numbers[i] + numbers[j];
                if (answer.Contains(res)) continue;
                else answer.Add(res);
            }
        }
        answer.Sort();
        return answer.ToArray();
    }
}